hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/handlers/OBSBucketHandler.java

```diff
@@ -0,0 +1,227 @@
+   * @return the total DU of all direct keys
+   * @throws IOException IOE
+   */
+  @Override
+  public long handleDirectKeys(long parentId, boolean withReplica,
+                               boolean listFile,
+                               List<DUResponse.DiskUsage> duData,
+                               String normalizedPath) throws IOException {
+
+    Table<String, OmKeyInfo> keyTable = getKeyTable();
+    long keyDataSizeWithReplica = 0L;
+
+    TableIterator<String, ? extends Table.KeyValue<String, OmKeyInfo>>
+        iterator = keyTable.iterator();
+
+    String seekPrefix = OM_KEY_PREFIX +
+        vol +
+        OM_KEY_PREFIX +
+        bucket +
+        OM_KEY_PREFIX;
+
+    NSSummary nsSummary = getReconNamespaceSummaryManager()
+        .getNSSummary(parentId);
+    // Handle the case of an empty bucket.
+    if (nsSummary == null) {
+      return 0;
+    }
+
+    iterator.seek(seekPrefix);
+
+    while (iterator.hasNext()) {
+      // KeyName : OmKeyInfo-Object
+      Table.KeyValue<String, OmKeyInfo> kv = iterator.next();
+      String dbKey = kv.getKey();
+
+      // Exit loop if the key doesn't match the seekPrefix.
+      if (!dbKey.startsWith(seekPrefix)) {
+        break;
+      }
+
+      OmKeyInfo keyInfo = kv.getValue();
+      if (keyInfo != null) {
+        DUResponse.DiskUsage diskUsage = new DUResponse.DiskUsage();
+        String objectName = keyInfo.getKeyName();
+        diskUsage.setSubpath(objectName);
+        diskUsage.setKey(true);
+        diskUsage.setSize(keyInfo.getDataSize());
+
+        if (withReplica) {
+          long keyDU = keyInfo.getReplicatedSize();
+          keyDataSizeWithReplica += keyDU;
+          diskUsage.setSizeWithReplica(keyDU);
+        }
+        // List all the keys for the OBS bucket if requested.
+        if (listFile) {
+          duData.add(diskUsage);
+        }
+      }
+    }
+
+    return keyDataSizeWithReplica;
+  }
+
+  /**
+   * Object stores do not support directories, hence return null.
+   *
+   * @return null
+   */
+  @Override
+  public long calculateDUUnderObject(long parentId)
+      throws IOException {
+    return Long.parseLong(null);
+  }
+
+  /**
+   * Object stores do not support directories, hence return null.
+   *
+   * @return null
+   */
+  @Override
+  public long getDirObjectId(String[] names) throws IOException {
+    return Long.parseLong(null);
+  }
+
+  /**
+   * Object stores do not support directories, hence return null.
+   *
+   * @return null
+   */
+  @Override
+  public long getDirObjectId(String[] names, int cutoff) throws IOException {
+    return Long.parseLong(null);
+  }
+
+  /**
+   * Returns the keyInfo object from the KEY table.
+   * @return OmKeyInfo
+   */
+  @Override
+  public OmKeyInfo getKeyInfo(String[] names) throws IOException {
+    String ozoneKey = OM_KEY_PREFIX;
+    ozoneKey += String.join(OM_KEY_PREFIX, names);
+
+    OmKeyInfo keyInfo = getKeyTable().getSkipCache(ozoneKey);
+    return keyInfo;
```

Comment:
nit: inline the variable `keyInfo`

```suggestion
    return getKeyTable().getSkipCache(ozoneKey);
```

Reply:
Thank you for the suggestion. I have implemented the recommended changes.